doc/src/advanced_tutorial/add_move_minimal.md

```diff
@@ -0,0 +1,236 @@
+Since it takes the system as parameter, we can compute quantities that may be important for the move - depending on the system we use.
+For our initial implementation though, we don't use the `setup` function.
+
+```rust
+fn setup(&mut self, _: &System) { 
+    // For now, do nothing.
+}
+```
+The `prepare` function is a bit more complicated:
+
+```rust
+fn prepare(&mut self, system: &mut System, rng: &mut Box<Rng>) -> bool {
+    // select a molecule randomly: pick a random number between 0 and the total number
+    // of molecules in the system.
+    let nmols = system.molecules().len();
+    if nmols == 0 {
+        return false // no molecule, so we can't perform the move
+    } else {
+        let molid = rng.gen_range(0, nmols)
+    }
+
+    // create a vector, containing the random displacements in every cartesian
+    // direction.
+    let delta = Vector3D::new(
+        self.range.sample(rng), // x
+        self.range.sample(rng), // y
+        self.range.sample(rng), // z
+    )
+
+    self.old_system = system.clone(); // before modifying the system, create a copy
+    // use the current positions and add the displacement
+    for i in self.old_system.molecule(molid) {
+        system[i].position += delta
+    }
+    true // we set everything up, so we can perform the move.
+}
+```
+
+That is a lot of stuff.
+Let's first have a look at the function arguments:
+We take three mutable references: One to the `Translate` struct (`&mut self`), one
+to the system and finally one to `Box<Rng>`, which is a **r**andom **n**umber **g**enerator.
+
+Let us now go through the code blocks one by one.
+
+First, we randomly select a molecule.
+All molecules are stored in a vector inside system, `system.molecules`.
+We cannot access this vector directly, since it is not public, but there is a public function that we can use: `system.molecules()` will return the vector of molecules.
+The `len()` function will return the number of elements in the vector which equals the number of molecules.
+To pick a random molecule, we draw a random number between 0 and the number of molecules.
+Note that we also check if there are any molecules in the system.
+
+Next, we create the random displacement.
+In Lumol, positions are stored as `Vector3D` structures.
+It is a vector with three elements, each of which stands for a cartesian coordinate.
+We create a random displacement vector by drawing a random number for every cartesian coordinate.
+In rust, we can do that by taking a sample from `range` which we set to be `[-dr, dr]` in our `new()` function.
+
+To apply the displacement, we first store the current state of the system using the `clone` function.
+In the `for` loop, we add the displacement by modifying the positions of the system.
+To understand what's going on here, we have to take a closer look on how positions are stored in Lumol and how we can access them:
+```rust
+for i in self.old_system.molecule(molid) {
```

Comment:
How about `for particle_index in system.molecule(molecule_index)`? It feels clearer to me.